src/main/java/edu/ucsb/cs156/dining/controllers/ReviewsController.java

```diff
@@ -0,0 +1,100 @@
+package edu.ucsb.cs156.dining.controllers;
+
+import edu.ucsb.cs156.dining.entities.User;
+import edu.ucsb.cs156.dining.entities.Reviews;
+import edu.ucsb.cs156.dining.models.CurrentUser;
+import edu.ucsb.cs156.dining.entities.UCSBDiningCommons;
+import edu.ucsb.cs156.dining.errors.EntityNotFoundException;
+import edu.ucsb.cs156.dining.repositories.ReviewsRepository;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import org.springframework.security.web.bind.annotation.AuthenticationPrincipal;
+
+import java.util.ArrayList;
+import jakarta.validation.Valid;
+
+/**
+ * This is a REST controller for Reviews
+ */
+
+@Tag(name = "Reviews")
+@RequestMapping("/api/reviews")
+@RestController
+@Slf4j
+public class ReviewsController extends ApiController {
+
+    @Autowired
+    ReviewsRepository reviewsRepository;
+
+    /**
+     * THis method returns a list of all reviews.
+     * @return a list of all reviews
+     */
+    @Operation(summary= "List all ucsb reviews of dining commons")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<Reviews> allReviews() {
+        Iterable<Reviews> reviews = reviewsRepository.findAll();
+        return reviews;
+    }
```

Comment:
What is this method for?   Where in the application do we need an endpoint that returns all reviews for ordinary users?

In fact, we do NOT want such a thing!  The whole point of the moderation is that users are NOT permitted to see all reviews; only reviews that have been approved by a moderator.

So, this method should, at the very least, be available to ADMIN users only.

These endpoints are clearly needed:
* One that is available to ROLE_USER and returns only all reviews authored by the current user.  This should not take the user id as a parameter, but instead, should ask for the id of the current user 
* One that is available to ROLE_USER that takes a menu item id as a parameter and returns all reviews that have moderation status "approved" AND match that menu id.  
* One that is available to ROLE_ADMIN that returns all reviews that are awaiting moderation.

What we do NOT want is this... one that returns all reviews to every user regardless of moderation status.